app/src/main/java/com/radixdlt/android/data/model/transaction/SendTokensLiveData.kt

```diff
@@ -23,11 +23,14 @@ class SendTokensLiveData @Inject constructor(
 ) : LiveData<String>() {
 
     private val compositeDisposable = CompositeDisposable()
+    private var balanceChecked = false
 
     fun sendToken(to: String, amount: BigDecimal, token: String, payLoad: String?) {
+        balanceChecked = false
         Identity.api!!.getBalance(Identity.api!!.getMyAddress())
             .subscribe {
-
+                if (balanceChecked) return@subscribe
+                balanceChecked = true
```

Comment:
This looks a bit funny to me. I would suggest a full context block for the early bail-out ("curly braces") or, if you are uncomfortable applying Java paradigms in Kotlin world, at least add an empty line after the if-block to clearly separate the `balanceChecked = true` from it.

Reply:
This this was a quick hack due to some weird behaviour with the .getBalance() which was proving to be a bit erratic... Will add a //FIX me comment and add a block for clarity.